Hazel/src/Hazel/Renderer/Renderer2D.cpp

```diff
@@ -42,30 +42,34 @@ namespace Hazel {
 		Renderer2D::Statistics Stats;
 	};
 
-	static Renderer2DData s_Data;
+	static Renderer2DData* s_Data = nullptr;
 
 	void Renderer2D::Init()
 	{
 		HZ_PROFILE_FUNCTION();
 
-		s_Data.QuadVertexArray = VertexArray::Create();
+		HZ_CORE_ASSERT(!s_Data, "Renderer2D is already initialized!");
 
-		s_Data.QuadVertexBuffer = VertexBuffer::Create(s_Data.MaxVertices * sizeof(QuadVertex));
-		s_Data.QuadVertexBuffer->SetLayout({
+		s_Data = new Renderer2DData();
```

Comment:
This suggestion looks weird... I meant to suggest to delete the empty line on line 52, such that the assert & the new Render2DData allocation are grouped as a block...
```suggestion
		HZ_CORE_ASSERT(!s_Data, "Renderer2D is already initialized!");
		s_Data = new Renderer2DData();
```